ydb/query/session.py

```diff
@@ -0,0 +1,311 @@
+        )
+
+        first_response = next(status_stream)
+        if first_response.status != issues.StatusCode.SUCCESS:
+            pass
+
+        self._state.set_attached(True)
+        self._state._change_state(QuerySessionStateEnum.CREATED)
+
+        threading.Thread(
+            target=self._check_session_status_loop,
+            args=(status_stream,),
+            name="check session status thread",
+            daemon=True,
+        ).start()
+
+    def _check_session_status_loop(self, status_stream):
+        try:
+            for status in status_stream:
+                if status.status != issues.StatusCode.SUCCESS:
+                    self._state.reset()
+                    self._state._change_state(QuerySessionStateEnum.CLOSED)
+        except Exception:
+            pass
+
+    def delete(self) -> None:
+        """
+        Deletes a Session of Query Service on server side and releases resources.
+
+        :return: None
+        """
+        if self._state._already_in(QuerySessionStateEnum.CLOSED):
+            return
+
+        self._state._check_invalid_transition(QuerySessionStateEnum.CLOSED)
+        self._delete_call()
+        self._stream.cancel()
+
+    def create(self) -> "QuerySessionSync":
+        """
+        Creates a Session of Query Service on server side and attaches it.
+
+        :return: QuerySessionSync object.
+        """
+        if self._state._already_in(QuerySessionStateEnum.CREATED):
+            return
+
+        self._state._check_invalid_transition(QuerySessionStateEnum.CREATED)
+        self._create_call()
+        self._attach()
+
+        return self
+
+    def transaction(self, tx_mode: base.BaseQueryTxMode = None) -> base.IQueryTxContext:
+        """
+        Creates a transaction context manager with specified transaction mode.
+        :param tx_mode: Transaction mode, which is a one from the following choises:
+         1) QuerySerializableReadWrite() which is default mode;
+         2) QueryOnlineReadOnly(allow_inconsistent_reads=False);
+         3) QuerySnapshotReadOnly();
+         4) QueryStaleReadOnly().
+
+        :return transaction context manager.
+
+        """
+        self._state._check_session_ready_to_use()
+
+        return BaseTxContext(
+            self._driver,
+            self._state,
+            self,
+            tx_mode,
+        )
+
+    def execute(
+        self,
+        query: str,
+        tx_mode: base.BaseQueryTxMode = None,
+        syntax: base.QuerySyntax = None,
+        exec_mode: base.QueryExecMode = None,
+        parameters: dict = None,
+        concurrent_result_sets: bool = False,
+        empty_tx_control: bool = False,
+    ):
+        """
+        Sends a query to Query Service
+        :param query: (YQL or SQL text) to be executed.
+        :param tx_mode: Transaction mode, which is a one from the following choises:
+         1) QuerySerializableReadWrite() which is default mode;
```

Comment:
for execute without explicit transaction mode it will not be QuerySerializableReadWrite, because you can't create table within QuerySerializableReadWrite mode